### PR TITLE
Bump to Core v0.15

### PR DESCRIPTION
**Pull request by ericthewry**
Following this blog post: https://blog.janestreet.com/goodbye-Core_kernel/

The following mapping largely summarizes the change.
`Core_kernel` -> `Core`
`Command.run` -> `Command_unix.run`
`Sys` -> `Sys_unix`

I also modified the `libraries` clause of the `dune` files, and made some slight semantic changes since `Sys_unix.file_exists` differs slightly from the deprecated `Sys.file_exists`.

### CODE REVIEW

**Review by jnfoster — APPROVED** [review]

**Comment by hackedy** [comment]
Half the CI builds were failing because core 0.15 requires ocaml >= 4.11, so I tried to fix that. Once those go through I'll merge

**Comment by hackedy** [comment]
Let's not merge this until https://github.com/janestreet/ocaml_intrinsics/issues/4 is fixed

**Pull request merged**
